service/main.toit

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2025 Toit contributors.
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file.
+
+import encoding.tison
+import system.assets
+import max31865.provider
+import max31865 show Driver
+
+install-from-args_ args/List:
+  if args.size != 3 and args.size != 4:
+    throw "Usage: main <clock> <miso> <mosi> [<cs>]"
+  clock := int.parse args[0]
+  miso := int.parse args[1]
+  mosi := int.parse args[2]
+  cs/int? := (args.size == 4) ? (int.parse args[3]) : null
+  provider.install --clock=clock --miso=miso --mosi=mosi --cs=cs
+
+install-from-assets_ configuration/Map:
+  clock := configuration.get "clock"
+  if not clock: throw "No 'clock' found in assets."
+  if clock is not int: throw "Clock must be an integer."
```

Comment:
I would probably try to reword the 'non-int' throws so the parameter name doesn't have to be capitalized.